engine/cache.js

```diff
@@ -1,24 +1,82 @@
-import fs from 'fs';
-import path from 'path';
+// import fs from 'fs';
+// import path from 'path';
+// import fetch from 'node-fetch';
+// import fileExists from 'file-exists';
+// import crypto from 'crypto';
+
+import Eu from 'eu';
+import redis from 'redis';
 import fetch from 'node-fetch';
-import fileExists from 'file-exists';
-import crypto from 'crypto';
+import stream from 'stream';
 
-export default {
-  readJson: (src, cacheDir) => {
-    const shasum = crypto.createHash('sha1');
-    shasum.update(src);
-    const basename = shasum.digest('hex');
-    const cacheFilename = path.join(cacheDir, `${basename}.json`);
-    if (fileExists(cacheFilename)) {
-      return Promise.resolve(JSON.parse(fs.readFileSync(cacheFilename)));
+function readJson(url, fake, dbNumber) {
+  console.log(`TODO remove cacheDir from readJson ${fake}`);
+  return new Promise((resolve, reject) => {
+    const client = redis.createClient(process.env.REDIS_URL, { no_ready_check: true });
+    if (!dbNumber) {
+      resolve(client);
+      return;
     }
-    return fetch(src)
-      .then((response) => response.text())
-      .then((text) => {
-        console.log(`Caching data for: "${src}" in "${cacheFilename}"`);
-        fs.writeFileSync(cacheFilename, text);
-        return JSON.parse(text);
+    client.select(dbNumber || 0, err => {
+      if (err) {
+        client.quit();
+        reject(err);
+        return;
+      }
+      resolve(client);
+    });
+  })
+  .then(client => {
+    const store = new Eu.RedisStore(client);
+    const cache = new Eu.Cache(store);
+    const eu = new Eu(cache);
+    return new Promise((resolve, reject) => {
+      eu.get(url, { json: true }, (err, res, body) => {
+        if (err) {
+          reject(err);
+          return;
+        }
+        const s = new stream.Readable();
+        s._read = function noop() {}; // redundant? see update below
+        s.push(JSON.stringify(body));
+        s.push(null);
+        return new fetch.Response(s, {
+          url,
+          status: res.statusCode,
+          headers: res.headers,
+        }).json()
+        .catch(jsonError => {
```

Comment:
Is this just to avoid eslint errors? If we just rethrow we can hand it down with the returned promise.